Documentation/install/upgrade.rst

```diff
@@ -7,12 +7,165 @@ Upgrade Guide
+
+
+In the `ConfigMap` above, we can verify that Cilium is using ``debug`` with
+``true``, it has a etcd endpoint running with `TLS <https://coreos.com/etcd/docs/latest/op-guide/security.html>`_,
+and the etcd is set up to have `client to server authentication <https://coreos.com/etcd/docs/latest/op-guide/security.html#example-2-client-to-server-authentication-with-https-client-certificates>`_.
+
+2. Download the `ConfigMap` with the changes for "\ |SCM_BRANCH|":
+
+.. tabs::
+  .. group-tab:: K8s 1.7
+
+    .. parsed-literal::
+
+      $ wget \ |SCM_WEB|\/examples/kubernetes/1.7/cilium-cm.yaml
+
+  .. group-tab:: K8s 1.8
+
+    .. parsed-literal::
+
+      $ wget \ |SCM_WEB|\/examples/kubernetes/1.8/cilium-cm.yaml
+
+  .. group-tab:: K8s 1.9
+
+    .. parsed-literal::
+
+      $ wget \ |SCM_WEB|\/examples/kubernetes/1.9/cilium-cm.yaml
+
+  .. group-tab:: K8s 1.10
+
+    .. parsed-literal::
+
+      $ wget \ |SCM_WEB|\/examples/kubernetes/1.10/cilium-cm.yaml
+
+  .. group-tab:: K8s 1.11
+
+    .. parsed-literal::
+
+      $ wget \ |SCM_WEB|\/examples/kubernetes/1.11/cilium-cm.yaml
+
+
+Verify its contents:
+
+.. literalinclude:: ../../examples/kubernetes/1.7/cilium-cm.yaml
+
+
+3. Add the new options manually to your old `ConfigMap`, and make the necessary
+changes.
+
+In this example, the ``debug`` option is meant to be kept with ``true``, the
+``etcd-config`` is kept unchanged, and ``legacy-host-allows-world`` is a new
+option, but after reading the :ref:`upgrade_notes` the value was kept unchanged
+from the default value.
+
+After making the necessary changes the old `ConfigMap` was migrated with the
+new options while keeping the configuration that we wanted:
+
+::
+
+        $ cat ./cilium-cm-old.yaml
+        apiVersion: v1
+        data:
+          debug: "true"
+          disable-ipv4: "false"
+          # If you want to clean cilium state; change this value to true
+          clean-cilium-state: "false"
+          legacy-host-allows-world: "false"
+          etcd-config: |-
+            ---
+            endpoints:
+            - https://192.168.33.11:2379
+            #
+            # In case you want to use TLS in etcd, uncomment the 'ca-file' line
+            # and create a kubernetes secret by following the tutorial in
+            # https://cilium.link/etcd-config
+            ca-file: '/var/lib/etcd-secrets/etcd-ca'
+            #
+            # In case you want client to server authentication, uncomment the following
+            # lines and add the certificate and key in cilium-etcd-secrets below
+            key-file: '/var/lib/etcd-secrets/etcd-client-key'
+            cert-file: '/var/lib/etcd-secrets/etcd-client-crt'
+        kind: ConfigMap
+        metadata:
+          creationTimestamp: null
+          name: cilium-config
+          selfLink: /api/v1/namespaces/kube-system/configmaps/cilium-config
+
+After adding the options manually save the file with your changes and install
```

Comment:
"After adding the options, manually save..."

Reply:
Done